### ISSUE TITLE
[bug] make panic error when go version is 1.22

### ISSUE THREAD

**Issue by kissycn** [kubevela/pkg#108]
After the version is upgraded to 1.22, make prompts a panic error. The main reason is a bug in controller-gen. 
[Bug Address](https://github.com/kubernetes-sigs/controller-tools/issues/880)
This bug requires upgrading controller-tools to 1.44, but this will cause a series of unknown risks, such as the k8s version will be upgraded to 1.29, controller-runtime will also be affected, etc.

**Comment by FogDong** [comment]
Thank you for the report. I believe our controller-tools is quite outdated and may need an update.
The Kubernetes version also needs to be updated, but currently, the community is short-handed. Would you be interested in taking on the task of upgrading these components?

**Comment by kissycn** [comment]
@FogDong 
Yes, I am trying to upgrade the k8s component version to 0.29.x, but there is a more serious problem: the apiserver-runtime of the cluster-gateway component only supports k8s stability up to 0.26.x and [0.27.x](https://github.com/kubernetes-sigs/apiserver-runtime/issues/93) has not been verified. Currently, the apiserver-runtime is stagnant. However, cluster-gateway is a core component, and other templates will depend on it, so let's see if the community has other solutions. Let's discuss it together.

**Comment by phantomnat** [comment]
> the apiserver-runtime is stagnant

I saw the PR in apiserver-runtime waiting for an approval so I used that commit.

Here are my 2 PRs.
- https://github.com/kubevela/pkg/pull/109
- https://github.com/oam-dev/cluster-gateway/pull/152
